pandas/core/frame.py

```diff
@@ -8869,11 +8869,30 @@ def update(
         if not isinstance(other, DataFrame):
             other = DataFrame(other)
 
-        other = other.reindex(self.index)
+        indexes_intersection = other.index.intersection(
+            self.index
+        )  # order is important
+        if not len(indexes_intersection):
+            raise ValueError(
+                "Can't update dataframe when other has no index in common with "
+                "this dataframe."
+            )
+
+        if other.index.is_unique:
```

Comment:
[I've recommended](https://github.com/pandas-dev/pandas/pull/55634#discussion_r1375265143) to raise in this case. Do you disagree?

Reply:
I don't understand my question here - looks like I might have read the code as being `if not other.index.is_unique`.